### DISCUSSION TITLE
How does KCL implement the compilation to both wasm32-unknown-unknown and wasm32-wasi targets?

### DISCUSSION THREAD

**Discussion by sv158** [kcl-lang/kcl#1767] [Q&A] [answered]
I've retrieved some relevant source code fragments, but I didn't get the basic method and the complete workflow, so I'm looking for tips on how to realize it. As a newbie to `rust-wasm`, what I've seen at this point is the following (not including their relationships):

1. some modules directly depend on `wasm-bindgen`;
2. `wasi` is present in some of the source code, makefile and cargo.lock; 
3. `codegen` can be found, but it's not clear how it relates to wasm configuration.

**Comment by Peefy** [comment]
cc @zong-zhe @He1pa

**Comment by He1pa** [comment, marked as answer]
Have you looked at https://github.com/kcl-lang/lib/tree/main/wasm
https://github.com/kcl-lang/lib/pull/137
